### DISCUSSION TITLE
Displaying step count on screen

### DISCUSSION THREAD

**Discussion by mattblank11** [agencyenterprise/react-native-health#123] [Q&A]
Hey team!

This may be an extremely dumb question, but how do you display step counts on the screen?

I'm able to use `AppleHealthKit.getStepCount` to get the step count, but if I try to return the step count and display it on the screen it returns a null value. I even tried async / await to no avail.

I'm a Python dev who is trying to build a really crude app that gets step counts but feel like a giant dummy here...

Thanks so much!
Matt

**Comment by lucaspbordignon** [comment]
Hey @mattblank11! Welcome to the project 👋🏾 

Regarding the question, you just need to make sure that you're requesting the user's permission before accessing the data. You can read more about that [here](https://github.com/agencyenterprise/react-native-health/blob/master/docs/permissions.md)

In your case, you need the `AppleHealthKit.Constants.Permissions.StepCount`. Could you check that?

Also, if that still doesn't work, we have a method that captures step count samples from multiple dates, called [getDailyStepCountSamples()](https://github.com/agencyenterprise/react-native-health/blob/master/docs/getDailyStepCountSamples.md)

In the case of the second works, we might have a potential bug on `getStepCount()`

**Reply by mattblank11** [reply]
Thanks @lucaspbordignon!

I figured out the issue - I needed to use hooks to change the state of a variable containing my step count so that I could display it on the screen. If there's a different solution, I'm all ears!

Appreciate the help!
